docs/threshold_encryption.md

```diff
@@ -0,0 +1,36 @@
+# Threshold Encryption
+*Threshold encryption* is one of cryptosystem, if in order to decrypt an encrypted message or to sign a message, several 
+parties (more than some threshold number) must cooperate in the decryption or signature protocol.
+The message is encrypted using a `public key` and the corresponding `private key` is shared among the participating parties.
+Only by possessing more than a threshold number of bits of the secret can the secret be determined.
+
+### Idea
+The idea of threshold cryptography is to protect information (or computation) by fault-tolerance distributing it among a
+cluster of cooperating computers. First consider the fundamental problem of threshold cryptography, a problem of secure
+sharing of a secret. A secret sharing scheme allows one to distribute a piece of secret information among several servers
+in a way that meets the following requirements.
+1. No group of corrupt servers (smaller than a given threshold) can figure out what the secret is, even if they cooperate.
+2. When it becomes necessary that the secret information be reconstructed, a large enough number of servers (a number
+larger than the above threshold) can always do it.
+
+### Example
+If I had four copies of a key and cut each key into three pieces, distributing one piece each to twelve people then it 
+would probably take about five people to use the key. The minimum threshold in this case would be three, and the maximum
```

Comment:
I think `have` is more smooth for this case, and could i get more explain for why five people is needed to use the key?

Reply:
It said, *probably take about five people*, this is just example.
In next sentence, *The minimum threshold in this case would be three*, it is what you think.

Reply:
Let's talk about this example.
Docs say there is one key, and copy it to 4 keys, and cut them to 3 pieces each.
Then, we have **12 pieces of key.**
If we recover original key, we should meet about 5~6, or more people.
But we can recover just meeting 3 people, who have 1, 2, 3 piece of key.

Reply:
Oh, i got it! Thank you for explain:)